### ISSUE TITLE
Command line flags don't override playbook values.

### ISSUE THREAD

**Issue by xamox** [ansible/ansible#3738]
If I do something like:

```
ansible-playbook install-software.yml -i "your.host.name," --user=ubuntu --extra-vars=user=ubuntu
```

For this playbook:

```
- hosts: all
  connection: ssh
  user: sm
  sudo: yes
  sudo_user: root
  gather_facts: true
  vars:
    - user: sm
    - code_directory: Code

  tasks:
  - name: update package list
    action: apt update_cache=yes
```

It still tries to use the user 'sm' instead of overriding with 'ubuntu'.

**Comment by mpdehaan** [comment]
This is working as intended.

You defined a variable called "user", but it means nothing to Ansible.

You can set ansible_ssh_user if you so choose on a per host basis (I think that's not a great idea), but you can also do this:

```
user: "{{ user }}"
```

**Comment by vladimir-sol** [comment]
@mpdehaan Excuse me for posting into an old thread, but I just ran into this with `-u` flag with a playbook like

```
---
- hosts: "{{ hosts }}"
  user: deployer
```

And with the invocation below

```
ansible-playbook this.yml -u root -e "hosts=$HOSTS"
```

I think it's reasonable to expect command line args to take precedence as it's consistent with passing variables with `-e` and in my opinion justifies existence of `-u` in the first place.

I ended up using

```
---
- hosts: "{{ hosts }}"
  user: "{{ user | default('deployer') }}"
```

which feels unnecessarily dirty.

Once again, my main concern is that command line option doesn't take precedence over what's written in playbook, which in my opinion is unnatural and damages flexibility.
